.github/workflows/swift.yml

```diff
@@ -1,40 +1,62 @@
 name: Swift
 
-on: [push]
-
-env:
-   CODECOV_TOKEN: ${{ secrets. CODECOV_TOKEN }}
+on: 
+  push:
+    branches: 
+    - master
```

Comment:
why just master?

Reply:
Because otherwise ci is automatically runs twice on each branch, that has a pull request, which IMHO is a waist of resources – even if it's free. 

I guess the normal way is as follows: one branch where the magic happens (master in this case). All other branches want to merge into this branch sooner or later. If one needs ci, one can always create a "Draft Request" for a branch that isn't fully baked and ready for a real PR.

Reply:
I'd much prefer we had ci on all branches though. It feels like I'm losing something here. Creating a PR just so you can get a ci build done seems to send the wrong messages about what a PR is.

Reply:
My idea was to minimize the number of runs the ci does, that are totally wasted resources. I have seen in the previous prs, travis has always run the exact test twice. If there is a branch that needs CI on every push, one can just add the branch to the ci file on that branch.

With that we are resourcefulness in the normal case but also have the freedom for more ci on push if needed. But in this case it's an opt in, whereas a wildcard branch doesn't even has the chance to opt out.

Reply:
I don't have a strong opinion here, so I think a tie should go to the incumbent :-)

I see the logic and appreciate the noble pursuit to be resourceful. I really wouldn't be too bothered if we went ahead with ci on PRs only. But I do think it's a nice convenience to have ci on each branch as a default. I find we most often test locally and push as infrequently as possible to branches on the remote repo for ci. So it's really not that much waste imo.

Reply:
fixed.